tools/snpeff.py

```diff
@@ -43,22 +44,22 @@ def version(self):
         return "4.1"
 
     def execute(self, command, args, JVMmemory=None, stdin=None, stdout=None):    # pylint: disable=W0221
-        if JVMmemory is None:
+        if not JVMmemory:
             JVMmemory = self.jvmMemDefault
 
         # the conda version wraps the jar file with a shell script
         if self.install_and_get_path().endswith(".jar"):
             tool_cmd = [
-                'java', '-Xmx' + JVMmemory, '-Djava.io.tmpdir=' + tempfile.tempdir, '-jar', self.install_and_get_path(),
+                'java', '-Xmx' + JVMmemory, '-Djava.io.tmpdir=' + tempfile.gettempdir(), '-jar', self.install_and_get_path(),
                 command
             ] + args
         else:
             tool_cmd = [
-                self.install_and_get_path(), '-Xmx' + JVMmemory, '-Djava.io.tmpdir=' + tempfile.tempdir, command
+                self.install_and_get_path(), '-Xmx' + JVMmemory, '-Djava.io.tmpdir=' + tempfile.gettempdir(), command
             ] + args
 
         _log.debug(' '.join(tool_cmd))
-        subprocess.check_call(tool_cmd, stdin=stdin, stdout=stdout)
+        return util.misc.run(tool_cmd)
```

Comment:
The end of annotate_vcf relies on being able to pass a file handle as the stdout named parameter, and it looks like this change may not accomodate that?